src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -202,33 +202,33 @@ void checkAndWarnIgnored(@Nullable Object propertyToIgnore, String propertyName)
   }
 
   /**
-   * This method tries to guess the runtime based on the appengine-web.xml of all services that are
-   * expected to run.
+   * This method determines if the dev app server should run with sandbox restrictions based on the
+   * runtime parsed from appengine-web.xml on all services. If one service can run with no sandbox
+   * restrictions, all other services will also run locally with no restrictions.
    *
    * @param services a list of app engine standard service directories
-   * @return {@code false} if only java7 modules are found or {@code true} is at least one java8
-   *     module is found (i.e. pure java8 or mixed java7/java8)
+   * @return {@code true} if there no need to enforce the sandbox restrictions.
```

Comment:
I think it's the opposite: true if necessary to enforce the sandbox restrictions.